meson.build

```diff
@@ -0,0 +1,467 @@
+    libgerbera_args += '-DHAVE_AVSTREAM_CODECPAR'
+  endif
+else
+  avcodec_dep = declare_dependency()
+  avfilter_dep = declare_dependency()
+  avformat_dep = declare_dependency()
+  avutil_dep = declare_dependency()
+endif
+
+if get_option('curl')
+  curl_dep = dependency('libcurl')
+  libgerbera_args += '-DHAVE_CURL'
+  libgerbera_args += '-DONLINE_SERVICES'
+  libgerbera_args += '-DATRAILERS'
+  libgerbera_args += '-DSOPCAST'
+else
+  curl_dep = declare_dependency()
+endif
+
+if get_option('exif')
+  exif_dep = dependency('libexif')
+  libgerbera_args += '-DHAVE_LIBEXIF'
+else
+  exif_dep = declare_dependency()
+endif
+
+if get_option('exiv2')
+  exiv2_dep = dependency('exiv2')
+  libgerbera_args += '-DHAVE_EXIV2'
+  if exiv2_dep.version().version_compare('<0.28')
+    libgerbera_args += '-D_LIBCPP_ENABLE_CXX17_REMOVED_AUTO_PTR'
+  endif
+else
+  exiv2_dep = declare_dependency()
+endif
+
+if get_option('ffmpegthumbnailer')
+  ffmpegthumbnailer_dep = dependency('libffmpegthumbnailer')
+  libgerbera_args += '-DHAVE_FFMPEGTHUMBNAILER'
+else
+  ffmpegthumbnailer_dep = declare_dependency()
+endif
+
+if get_option('inotify')
+  if host_machine.system() == 'linux'
+    libgerbera_args += '-DHAVE_INOTIFY'
+  else
+    error('inotify is Linux only. Please disable.')
+  endif
+endif
+
+if get_option('js')
+  duktape_dep = dependency('duktape')
+  libgerbera_args += '-DHAVE_JS'
+else
+  duktape_dep = declare_dependency()
+endif
+
+if get_option('lastfm')
+  lastfm_dep = dependency('lastfm')
+  libgerbera_args += '-DHAVE_LASTFMLIB'
+else
+  lastfm_dep = declare_dependency()
+endif
+
+if get_option('magic')
+  magic_dep = cpp.find_library('magic')
+  libgerbera_args += '-DHAVE_MAGIC'
+else
+  magic_dep = declare_dependency()
+endif
+
+if get_option('matroska')
+  ebml_dep = dependency('libebml')
+  matroska_dep = dependency('libmatroska')
+  libgerbera_args += '-DHAVE_MATROSKA'
+else
+  ebml_dep = declare_dependency()
+  matroska_dep = declare_dependency()
+endif
+
+if get_option('mysql')
+  mysql_dep = dependency('libmariadb')
+  libgerbera_args += '-DHAVE_MYSQL'
+else
+  mysql_dep = declare_dependency()
+endif
+
+if get_option('npupnp')
+  upnp_dep = dependency('libnpupnp')
+  libgerbera_args += '-DUSING_NPUPNP'
+else
+  upnp_dep = dependency('libupnp')
+endif
+
+if get_option('taglib')
+  taglib_dep = dependency('taglib')
+  libgerbera_args += '-DHAVE_TAGLIB'
+else
+  taglib_dep = declare_dependency()
+endif
+
+libgerbera_deps = [
+  avcodec_dep,
+  avfilter_dep,
+  avformat_dep,
+  avutil_dep,
+  curl_dep,
+  duktape_dep,
+  ebml_dep,
+  exif_dep,
+  exiv2_dep,
+  fs_dep,
+  fmt_dep,
+  ffmpegthumbnailer_dep,
+  iconv_dep,
+  lastfm_dep,
+  libatomic_dep,
+  magic_dep,
+  matroska_dep,
+  mysql_dep,
+  nsl_dep,
+  pugixml_dep,
+  socket_dep,
+  spdlog_dep,
+  sqlite3_dep,
+  taglib_dep,
+  upnp_dep,
+  uuid_dep,
+]
+
+# mangix hacks
+add_global_arguments('-DCOMPILE_INFO="mangix"', language : 'cpp')
```

Comment:
You probably want add_project_arguments instead, which does exactly the same thing unless subprojects are involved, in which case add_global_arguments accidentally infects all subprojects even though no one actually wants that -- hence why add_project_arguments got added. :D

Reply:
hence the hacks comment :).

They'll need to be implemented properly at some point to match CMake.